x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/ranges/range_editor.tsx

```diff
@@ -0,0 +1,237 @@
+              defaultMessage: 'Auto interval',
+            })}
+            checked={autoIntervalEnabled}
+            onChange={(e) => toggleAutoInterval(e.target.checked)}
+            data-test-subj="indexPattern-ranges-auto-interval"
+            compressed
+          />
+        }
+      >
+        <>
+          {autoIntervalEnabled ? (
+            <EuiRange
+              compressed
+              showLabels
+              showInput="inputWithPopover"
+              data-test-subj="lns-indexPattern-range-maxBars-field"
+              min={1}
+              max={maxHistogramBars}
+              step={1}
+              value={getMaxBarsValueToShow(maxBarsValue, maxHistogramBars)}
+              onChange={({ currentTarget }) => setMaxBarsValue(currentTarget.value)}
+              placeholder={i18n.translate('xpack.lens.indexPattern.ranges.autoIntervals', {
+                defaultMessage: 'Enter the max interval',
+              })}
+              prepend={
+                <>
+                  <EuiText size="s">
+                    {i18n.translate('xpack.lens.indexPattern.ranges.maxIntervals', {
+                      defaultMessage: 'Max intervals',
+                    })}
+                  </EuiText>{' '}
+                  <EuiIconTip
+                    position="right"
+                    content={i18n.translate('xpack.lens.indexPattern.ranges.maxIntervalsHelp', {
+                      defaultMessage:
+                        "Intervals will be selected automatically based on the available data. The maximum number of bars can never be greater than the Advanced Setting's {histogramMaxBars}",
+                      values: { histogramMaxBars: UI_SETTINGS.HISTOGRAM_MAX_BARS },
+                    })}
+                    type="questionInCircle"
+                  />
+                </>
+              }
+            />
+          ) : (
+            <EuiFieldNumber
+              compressed
+              data-test-subj="lns-indexPattern-range-interval-field"
+              value={intervalValue}
+              onChange={({ target }) => setIntervalValue(target.value)}
+              placeholder={i18n.translate('xpack.lens.indexPattern.ranges.enterInterval', {
+                defaultMessage: 'Enter an interval',
+              })}
+              prepend={
+                <>
+                  <EuiText size="s">
+                    {i18n.translate('xpack.lens.indexPattern.ranges.min', {
+                      defaultMessage: 'Min interval',
+                    })}
+                  </EuiText>{' '}
+                  <EuiIconTip
+                    position="right"
+                    content={i18n.translate('xpack.lens.indexPattern.ranges.minIntervalsHelp', {
+                      defaultMessage:
+                        "Interval will be automatically scaled in the event that the provided value creates more buckets than specified by Advanced Setting's {histogramMaxBars}",
```

Comment:
```suggestion
                        "Intervals will be automatically scaled if the provided value creates more buckets than specified by Advanced Setting's {histogramMaxBars}",
```

^^ Unless we can actually **display** the value of `histogramMaxBars` here instead? Like:

```suggestion
                        "Intervals will be automatically scaled if the provided value creates more than 100 buckets",
```